server/recceiver/cfstore.py

```diff
@@ -0,0 +1,207 @@
+# -*- coding: utf-8 -*-
+
+import itertools
+
+from zope.interface import implements
+from twisted.internet import defer
+from twisted.application import service
+from twisted.enterprise import adbapi as db
+from channelfinder import ChannelFinderClient
+import interfaces
+import datetime
+
+# ITRANSACTION FORMAT:
+#
+# src = source address
+# addrec = records ein added ( recname, rectype, {key:val})
+# delrec = a set() of records which are being removed
+# infos = dictionary of client infos
+# recinfos = additional infos being added to existing records 
+# "recid: {key:vale}"
+#
+
+
+__all__  = ['CFProcessor']
+
+class CFProcessor(service.Service):
+    implements(interfaces.IProcessor)
+
+    def __init__(self, name, conf):
+        print "CF_INIT"
+        self.name,self.conf = name,conf
+        print "CONF"+str(conf)
+        
+        
+    def startService(self):
+        service.Service.startService(self)
+        self.running = 1
+        print "CF_START"
+        try:
+            '''
+            Using the default python cf-client.
+            The usr, username, and password are provided by the channelfinder._conf module.
+            '''
+            self.client = ChannelFinderClient()
+        except:
+            print 'Failed to create cf client'
+        
+    def stopService(self):
+        service.Service.stopService(self)
+        #Set channels to inactive and close connection to client
+        self.running = 0
+        print "CF_STOP"
+        
+    def execute(self, cmd):
+        pass
+
+    def add(self, rec):
+        pass
+    
+    def delete(self, rec):
+        pass
+    
+    def commit(self, TR):
+        print "CF_COMMIT"
+#    print TR.src
+#    print TR.src.host, TR.src.port
+        print [(K,V) for K,V in TR.infos.iteritems()]
+        pvNames = [unicode(rname, "utf-8") for rid, (rname, rtype) in TR.addrec.iteritems()]
+        iocName=TR.src.port
+        hostName=TR.src.host
+        owner='cfstore'
+    
+        '''
+        Currently using the hostIP and the iocPort
+        if 'IOCNAME' in TR.infos:
+                iocName = TR.infos['IOCNAME']
+        if 'HOSTNAME' in TR.infos:
+                hostName = TR.infos['HOSTNAME']
+        '''
+        if 'ENGINEER' in TR.infos:
+            owner = TR.infos['ENGINEER']
```

Comment:
Presumable this will fail for nsls2 or frib as "mdavidsaver x3698" is not a valid user name?  This variable has traditionally been free-form, I'd suggest adding a new CF specific variable (eg. "CF_USERNAME").  At least don't fail hard if "ENGINEER" isn't a valid username.  Maybe retry with the default.